### PR TITLE
Add mint & burn to L2 ETH

### PR DESCRIPTION
**Pull request by karlfloersch**
## Questions
- 
-
-

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

**Comment by tynes** [comment]
Maybe consider adding `increaseAllowance` and `decreaseAllowance` to prevent frontrunning with `approve`. I believe this could also be taken advantage within a single transaction if an evil contract is called and transfers the full approve amount before  `approve` is called again later in the execution path

https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-decreaseAllowance-address-uint256-

File: contracts/optimistic-ethereum/OVM/precompiles/OVM_ETH.sol
Comment (K-Ho) [inline contracts/optimistic-ethereum/OVM/precompiles/OVM_ETH.sol:52]:
```suggestion
        require(msg.sender == address(_messenger), "Only callable by xDomainMessenger.");
         require(msg.sender == address(_messenger), "Only callable by OVM_L2CrossDomainMessenger.");
```

File: contracts/optimistic-ethereum/OVM/precompiles/OVM_ETH.sol
Comment (K-Ho) [inline contracts/optimistic-ethereum/OVM/precompiles/OVM_ETH.sol:53]:
```suggestion
        require(_messenger.xDomainMessageSender() == resolve("OVM_EthDepositContract"), "Only the L1 bridge can invoke.");
                require(_messenger.xDomainMessageSender() == resolve("OVM_EthDepositContract"), "Only the L1 OVM_EthDepositContract can invoke.");
```

Comment (ben-chain) [inline contracts/optimistic-ethereum/OVM/precompiles/OVM_ETH.sol:53]: worth specifying l1 in the name here?

**Comment by karlfloersch** [comment]
> Maybe consider adding `increaseAllowance` and `decreaseAllowance` to prevent frontrunning with `approve`. I believe this could also be taken advantage within a single transaction if an evil contract is called and transfers the full approve amount before `approve` is called again later in the execution path
> 
> https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-decreaseAllowance-address-uint256-

@tynes what do you think about holding off on this for now & adding it in a future release? mint/burn covers the basics of what we need for ETH deposits & so at this point trying to just focus on that

**Comment by tynes** [comment]
I'm fine with adding it in a future release, makes sense to not squeeze too many things in at once. Would be more useful when there is arbitrary contract deployment

**Review by K-Ho — DISMISSED** [review]
lgtm - need @ben-chain change to make mint and burn `onlyOwner`, but other than that looks great

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
LGTM

**Pull request merged**
